src/common/deflate.cpp

```diff
@@ -81,6 +87,55 @@ byte *decompressDeflate(const byte *data, size_t inputSize,
 	return decompressedData.release();
 }
 
+byte *decompressDeflateWithoutOutputSize(const byte *data, size_t inputSize, size_t &outputSize,
+						int windowBits, unsigned int frameSize) {
+	z_stream strm;
+	BOOST_SCOPE_EXIT( (&strm) ) {
+			inflateEnd(&strm);
+	} BOOST_SCOPE_EXIT_END
+
+	initZStream(strm, windowBits, inputSize, data);
+
+	std::vector<byte *> buffers;
+
+	int zResult = 0;
+	do {
+		buffers.push_back(new byte[frameSize]);
+
+		// Set the output data pointer and size
+		strm.avail_out = frameSize;
+		strm.next_out = buffers[buffers.size()-1];
```

Comment:
You can use [back()](http://en.cppreference.com/w/cpp/container/vector/back) instead.

Reply:
On the other hand, isn't that rather a list you have here?

Also, isn't that something for our PtrList class, with a DeallocatorArray?

I.e. it could be a `Common::PtrList<byte, DeallocatorArray>`.

Reply:
(In case you don't know, the PtrList provides a managed list with automatic deletion of elements. Something like a ScopedArray, but for a list of things. With the DeallocatorArray, you can put in arrays of things, i.e. things that need to be deallocated with delete[].)